### PR TITLE
importccl: refactor handling of IMPORT csv options

### PR DESCRIPTION
**Pull request by dt**
The main change here is to always pass the csv options (comma, nullif, etc) as the proton-defined bundle, rather than various ad-hoc combinations. We had a mixture of passing `roachpb.CSVOptions` and various re-creations of the same fields being used in signatures, structs and other protos before. This standardizes on always passing the proto-defined bundle of format options, while also generalizing it to allow specifying non-CSV formats and whatever format-specific options they require in the future.

Unfortunately this duplication has even spread into various persisted and rpc protos, so legacy compatibility requires keeping the old fields around for now and handling them.

Some uninteresting code movement too, namely breaking sst_writer into its own file since csv.go was getting messy.

### CODE REVIEW

File: pkg/ccl/importccl/csv.go [hunk 1]
```diff
@@ -871,15 +867,21 @@ func newReadCSVProcessor(
...
-		csvOptions: spec.Options,
-		sampleSize: spec.SampleSize,
-		tableDesc:  spec.TableDesc,
-		uri:        spec.Uri,
-		output:     output,
-		settings:   flowCtx.Settings,
-		registry:   flowCtx.JobRegistry,
-		progress:   spec.Progress,
+		flowCtx:     flowCtx,
+		inputFromat: spec.Format,
```
Comment (mjibson) [inline pkg/ccl/importccl/csv.go]: Fromat -> Format

File: pkg/sql/distsqlrun/processors.proto [hunk 2]
```diff
@@ -548,6 +548,8 @@ message JobProgress {
 // See ccs/sqlccl/csv.go for implementation.
 message ReadCSVSpec {
   optional roachpb.CSVOptions options = 1 [(gogoproto.nullable) = false];
```
Comment (mjibson) [inline pkg/sql/distsqlrun/processors.proto]: doesn't this line need to go away?

Comment (dt) [reply pkg/sql/distsqlrun/processors.proto]: No? I think we want to keep it so if a pre-refactor node sends a spec, which sets this field, a post-refactor node is able to unpack/repack into the nested version.

**Review by madelynnblue — APPROVED** [review]

**Pull request merged**